src/working/merge_sort.py

```diff
@@ -0,0 +1,76 @@
+"""Merge sort data structure."""
+
+
+def mergesort(data):
+    """Sorting algorithm for merg sort."""
+    try:
+        if len(data) > 1:
+            half = len(data) // 2
+            left = data[:half]
+            right = data[half:]
+
+            mergesort(left)
+            mergesort(right)
+
+            di = 0
+            li = 0
+            ri = 0
+
+            while len(left) > li and len(right) > ri:
+                if not isinstance(left[li], int):
+                    raise TypeError()
+                if not isinstance(right[ri], int):
+                    raise TypeError()
+                if left[li] < right[ri]:
+                    data[di] = left[li]
+                    li += 1
+                else:
+                    data[di] = right[ri]
+                    ri += 1
+                di += 1
+
+            while len(left) > li:
+                data[di] = left[li]
+                li += 1
+                di += 1
+
+            while len(right) > ri:
+                data[di] = right[ri]
+                ri += 1
+                di += 1
+
+        return data
+
+    except TypeError:
+        return 'This sort method only accepts values of one \
```

Comment:
Are you trying to raise an error here? This is just returning a string which is a little weird.